policy/modules/system/systemd.fc

```diff
@@ -70,6 +77,13 @@
 /run/\.nologin[^/]*	--	gen_context(system_u:object_r:systemd_sessions_runtime_t,s0)
 /run/nologin	--	gen_context(system_u:object_r:systemd_sessions_runtime_t,s0)
 
+/run/user/[^/]+/systemd	-d	gen_context(system_u:object_r:systemd_user_runtime_t,s0)
+/run/user/[^/]+/systemd/generator(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
+/run/user/[^/]+/systemd/generator\.early(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
+/run/user/[^/]+/systemd/generator\.late(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
+/run/user/[^/]+/systemd/transient(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
+/run/user/[^/]+/systemd/user(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
+
```

Comment:
I might be wrong but practically i only ever seen /run/user/*/systemd/transient being used for user runtime units

Reply:
scratch that. There are indeed generator dirs there. good call

By the way you might be able to user %{USERID} instead of `/[^/]+`

Reply:
`/run/user/UID/systemd/user`  seems not needed though and if it is then you did not address a type transitiion for it below

Reply:
Not sure if it's actually needed or not, but `systemd.unit(5)` lists this as a valid location for user units (among many others). I haven't seen these units myself yet but I figured I would add it for consistency. Looks like I just forgot to add the `type_transition` below, so I will add it there.